tools/interface/ontology.py

```diff
@@ -7,34 +7,37 @@
 from yamlformat.validator.presubmit_validate_types_lib import ConfigUniverse
 from yamlformat import constants
 
-StandardField = NamedTuple('StandardField', [('namespace', str), ('name', str), ('is_required', bool)])
-
-StandardSubfield = NamedTuple('StandardSubfield', [('namespace', str), ('name', str)])
-
 class Ontology(object):
     """Class defining an implementation of Google's ConfigUniverse as an ontology
 
+       Args:
+            universe: an instantiated ConfigUniverse Object
+
        Attributes:
-            self.universe = A ConfigUniverse object detailing the various universes in the ontology
+            universe = A ConfigUniverse object detailing the various universes in the ontology
+
+       Returns:
+            An instance of Ontology class
     """
 
     def __init__(self, universe: ConfigUniverse):
         super().__init__()
         self.universe = universe
 
-    def GetFieldsForTypeName(self, entity_type: EntityType, required_only: bool = False) -> List[StandardField]:
+    def GetFieldsForTypeName(self, namespace: str, entit_type_name: str, required_only: bool = False) -> List[EntityTypeField]:
         """gets a list of fields for a given typename within a namespace
 
            Args:
-                entity_type: An EntityType object to get fields for 
+                namespace: the name of the namespace as a string
+                entity_type_name: name of the entity type as a string
                 required_only: when true will return only required fields for a given type
 
            Returns:
                 result_fields: a list of StandardField tuples
         """
         pass
      
-    def GetEntitiesFromFields(self, field_list: List[StandardField], general_type: str = None) -> List[EntityType]:
+    def GetEntityTypesFromFields(self, field_list: List[StandardField], general_type: str = None) -> List[EntityType]:
```

Comment:
This can be an exercise for later, but it probably makes sense to have your own local version of an entitytype.